### ISSUE TITLE
Parameter expansion of default values is not following POSIX

### ISSUE THREAD

**Issue by D-Brox** [fizyr/subst#19]
The syntax `${variable:default}` is not one of the 8 valid cases described on the POSIX specifications:

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

**Comment by de-vri-es** [comment]
That's true. This crate is "shell-like" substitution. It does not aim to be POSIX compatible.

**Comment by D-Brox** [comment]
True, it's `shell-like`, but it would still be desirable if at least `${variable:-default}` was also valid.

**Comment by de-vri-es** [comment]
I'm sorry, but this has never been a goal of the library. Being compatible with POSIX is much more complicated than what this crate aims to be.

**Comment by D-Brox** [comment]
No problem, I can work around this.

Can you take a look at #18 later? It has both an enhancement (permissive and partial substitution modes) and a fix for nested substitutions.

**Comment by de-vri-es** [comment]
Yeah. I've been busy in a move lately, but I will try to get to it :)
